composites/AlgoliaSearch/SearchBar.js

```diff
@@ -1,47 +1,168 @@
 import React from "react";
 import PropTypes from "prop-types";
+import styled from "styled-components";
+import { intlShape, injectIntl, defineMessages } from "react-intl";
+import debounce from "lodash/debounce";
 
-/**
- * Overrides the default submit action with a custom one.
- *
- * @param {Event} submitEvent The submit event.
- * @param {object} props The React props.
- * @returns {void}
- */
-const onSubmit = ( submitEvent, props ) => {
-	submitEvent.preventDefault();
-	props.submitAction( submitEvent );
+import colors from "../../style-guide/colors.json";
+import SearchIcon from "../../style-guide/svg/search.svg";
+import { Icon } from "../Plugin/Shared/components/Icon";
+
+const messages = defineMessages( {
+	headingText: {
+		id: "searchbar.headingtext",
+		defaultMessage: "Search the Yoast knowledge base",
+	},
+} );
+
+const SearchLabel = styled.label`
+	width: 2em;
+	height: 2em;
+	float: left;
+	margin-top: 0.25em;
+`;
+
+SearchLabel.propTypes = {
+	className: PropTypes.string,
+	htmlFor: PropTypes.string,
 };
 
+SearchLabel.defaultProps = {
+	className: "",
+	htmlFor: "",
+};
+
+const SearchBarWrapper = styled.div`
+	overflow: hidden;
+	width: 100%;
+`;
+
+const SearchBarInput = styled.input`
+	box-sizing: border-box;
+	height: 1em;
+	width: calc(100% - 2em);
+	box-shadow: inset 0 2px 8px 0px rgba(0,0,0,0.3);
+	background: ${ colors.$color_grey_light };
+	padding: 20px;
+	border: 0;
+	font-size: 1em;
+	float: left;
+`;
+
+const SearchHeading = styled.h2`
+	font-size: 1em;
+	margin: 0.5em 0;
+`;
+
 /**
  * Create the JSX to render the search bar.
  *
  * @param {object} props The React props.
  * @returns {ReactElement} A div with the searchbar.
  * @constructor
  */
-const SearchBar = ( props ) => {
-	return (
-		<div className="wpseo-kb-search-search-bar">
-			<h2 id="wpseo-kb-search-heading">{ props.headingText }</h2>
-			<form onSubmit={ ( submitEvent ) => onSubmit( submitEvent, props ) }>
-				<input type="text" aria-labelledby="wpseo-kb-search-heading" defaultValue={ props.searchString }/>
-				<button type="submit" className="button wpseo-kb-search-search-button">{ props.searchButtonText }</button>
-			</form>
-		</div>
-	);
-};
+class SearchBar extends React.Component {
+
+	/**
+	 * Constructs the component and sets its initial state.
+	 *
+	 * @param {Object} props The props to use for this component.
+	 *
+	 * @constructor
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			doRequest: false,
+			searchString: "",
+		};
+	}
+
+	/**
+	 * Defines the debounced doFormSubmission function to be triggered after the user quits typing for 1 second.
+	 *
+	 * @returns {void}
+	 */
+	componentWillMount() {
+		this.doFormSubmission = debounce( ( searchString ) => {
+			this.props.submitAction( searchString );
+		}, 1000 );
```

Comment:
I think this debounced call should be canceled also when the component unmounts. Right now, it gets canceled only when submitting, see `this.doFormSubmission.cancel();`